src/apps/service/apps/de/edit.clj

```diff
@@ -547,21 +582,28 @@
         (remove-nil-vals))))
 
 (defn copy-app
-  "This service makes a copy of an App available in Tito for editing."
+  "This service makes a copy of an app available for editing."
   [user app-id]
   (let [app (persistence/get-app app-id)]
     (verify-app-permission user app "read")
     (add-app user (convert-app-to-copy app))))
 
+(defn copy-app-version
+  "This service makes a copy of a specific app version available for editing."
+  [user app-id version-id]
+  (let [app (persistence/get-app-version app-id version-id)]
+    (verify-app-permission user app "read")
+    (add-app user (convert-app-to-copy app))))
```

Comment:
Good catch. I added that `versions` listing to the `ui` response (which is re-used here for formatting copies) after I created the `copy-app-version` function, so I'll have to test the copy endpoints again; but I think it should be OK since the functions that add fields to the db usually select only the fields needed for that function.

For example:
https://github.com/cyverse-de/apps/blob/71a2f3166e07f75f7a4d8f70a0a2b56ecbd9241a/src/apps/persistence/app_metadata.clj#L347-L362

It still might be a good idea to dissoc `versions` as well, to prevent potential future problems.